src/erasure-code/ErasureCode.cc

```diff
@@ -60,6 +60,28 @@ int ErasureCode::minimum_to_decode(const set<int> &want_to_read,
   return 0;
 }
 
+int ErasureCode::minimum_to_decode2(const set<int> &want_to_read,
+                                   const set<int> &available_chunks,
+                                   map<int, list<pair<int, int>>> *minimum)
+{
+  set<int> minimum_shard_ids;
+  int r = minimum_to_decode(want_to_read, available_chunks, &minimum_shard_ids);
+
+  if ( r != 0) return r;
```

Comment:
please follow CodingStyle... in this case, no blank like after 'int r = ...', no space after 'if (', braces and newlines around 'return r'